browser-tests/event-search/eventSearchPage.components.ts

```diff
@@ -0,0 +1,279 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import { screen, within } from '@testing-library/testcafe';
+import TestController from 'testcafe';
+
+import translations from '../../src/common/translation/i18n/fi.json';
+import { getEventFields } from '../../src/domain/event/EventUtils';
+import { KeywordOption } from '../../src/domain/event/types';
+import { formatDate } from '../../src/util/dateUtils';
+import getDateRangeStr from '../../src/util/getDateRangeStr';
+import toPascalCase from '../../src/util/toPascalCase';
+import { getErrorMessage } from '../utils/error.util';
+import { getExpectedEventContext } from '../utils/event.utils';
+import {
+  EventFieldsFragment,
+  Neighborhood,
+  PlaceFieldsFragment,
+} from '../utils/generated/graphql';
+import { regExpEscaped } from '../utils/regexp.util';
+
+export const getEventSearchPageComponents = (t: TestController) => {
+  const searchContainer = () => {
+    const selectors = {
+      component() {
+        return screen.findByTestId('searchContainer');
+      },
+      withinComponent() {
+        return within(screen.getByTestId('searchContainer'));
+      },
+      searchButton() {
+        return this.withinComponent().findByRole('button', { name: 'Hae' });
+      },
+      dateRangeFilter() {
+        return this.withinComponent().findByLabelText('Valitse ajankohta');
+      },
+      dateRangeCheckbox(range: string) {
+        return this.withinComponent().findByRole('checkbox', {
+          name: range,
+        });
+      },
+      neighborhoodFilter() {
+        return this.withinComponent().findByLabelText('Etsi alue');
+      },
+      neighborhoodCheckbox(n: Neighborhood) {
+        return this.withinComponent().findByRole('checkbox', {
+          name: n.name.fi,
+        });
+      },
+      placeFilter() {
+        return this.withinComponent().findByLabelText('Etsi tapahtumapaikka');
+      },
+      placeSearchInput() {
+        return this.withinComponent().findByLabelText('Kirjoita hakusana');
+      },
+      placeCheckbox(place: PlaceFieldsFragment) {
+        return this.withinComponent().findByRole('checkbox', {
+          name: place.name.fi,
+        });
+      },
+      searchInput() {
+        return this.withinComponent().findByPlaceholderText(
+          'Kirjoita hakusana, esim. rock tai jooga'
+        );
+      },
+    };
+    const expectations = {
+      async isPresent() {
+        await t.expect(selectors.component().exists).ok(getErrorMessage(t));
+      },
+      async neighborhoodOptionIsPresent(neighborhood: Neighborhood) {
+        await this.isPresent();
+        t.ctx.neighborhood = neighborhood;
```

Comment:
I'm not quite sure where these are used? Are some of them redundant or just added in case they are used later?

Reply:
Oh ok they are used in the error message :)

Reply:
The default error message wont tell much. However now it does not print html anymore, I still have to improve the error message

Reply:
Now it prints always the "within" component:
https://github.com/City-of-Helsinki/events-helsinki-ui/pull/184/files#diff-70f4d2d9a0ec9b325b6d84d4c1601f1dfb7e8cf3d0dfb68b93d6c54107cc66c1R4-R22